hashintel/hash
# Fixing no-default-features

## 🌟 What is the purpose of this PR?

Previously passing `--no-default-features` to `cargo` commands failed to disable the features of our crates. This PR fixes this, and therefore simplifies the `cargo.toml` and developer experience on intel-based macs. 

## 🔗 Related links




- [Asana task description](https://app.asana.com/0/1199550852792314/1201495591878530/f)

## 🔍 What does this change?

The flag didn't work because it only applies to the root cargo project, and in cargo it's not possible to affect a dependency's dependency's features (two levels of nesting). Therefore, dependencies specified like `hash_engine = { path = "../.." }` would lead to the `hash_engine` project being built _with_ default features on, regardless of the flag being passed.

As such, those dependencies have had their default features explicitly disabled, and instead those features are manually set as default features at a higher level, which the flag _does_ propagate to.

The PR also updates the README to reflect the new instructions, where only ARM-based mac developers now need to do something bespoke.

## 🛡 Tests

- ✅ Manual Tests

## ❓ How to test this?

1. Checkout the branch
2. Confirm it compiles with the instructions in the README
